.github/workflows/slow-tests.yml

```diff
@@ -60,14 +60,16 @@ jobs:
       - uses: ./.github/actions/free-disk-space
       - uses: ./.github/actions/untracked
       - uses: ./.github/actions/setup-python
+        with:
+          python-version: 3.9
       - uses: ./.github/actions/setup-pyenv
       - uses: ./.github/actions/setup-java
       - name: Install dependencies
         run: |
           python -m venv .venv
           source .venv/bin/activate
           source ./dev/install-common-deps.sh --ml
-          python -m pip install langchain_experimental
+          python -m pip install langchain_experimental tf-keras
```

Comment:
why do we need tf-keras?

Reply:
without this, this error occurs: https://github.com/WeichenXu123/mlflow/actions/runs/9647331976/job/26605841779#step:10:74

Reply:
what triggered this failure?

Reply:
keras3 is installed after upgrading to python 3.9,

previously in python 3.8, it installs Collecting keras<2.13,>=2.12.0 (from tensorflow>=2.10.0->-r requirements/extra-ml-requirements.txt (line 12))

Reply:
Keras 3 only supports python >= 3.9, so upgrading to python 3.9 triggers this.

Reply:
Got it. Thanks for the investigation. It'd be nice if you could leave a review comment next time for a change like this so reviewers don't need to guess.